### ISSUE TITLE
Rule Change: Update error loc in max-statements to function header

### ISSUE THREAD

**Issue by kimtaejin3** [eslint/eslint#20932]
### What rule do you want to change?

max-statements

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
/*eslint max-statements: ["error", 2]*/

function foo() {
    let bar = 1;
    let baz = 2;
    let qux = 3;
}
```

### What does the rule currently do for this code?

Reports the error spanning the entire function body. For long functions, this can be visually disruptive in editors, where the error highlight covers many lines.

### What will the rule do after it's changed?

 Reports the error only on the function header (e.g., `function foo()`), matching the behavior of `max-params`.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [ ] I did not use AI to generate this issue report.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

Apologies for opening the PR (#20907) first without following the [guide](https://eslint.org/docs/latest/contribute/pull-requests#get-started)
The PR has been marked as draft by @lumirlumir. I'll wait for consensus before moving forward.

**Comment by DMartens** [comment]
Thank you for filing an issue for this.
As already said on the PR I am in favor of this proposal as we recently updated the `complexity` report location similarly.

**Comment by kecrily** [comment]
I think that's great; we've already reached a consensus on similar issues.

**Comment by mdjermanovic** [comment]
Makes sense to me, marking as accepted.
